noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -1339,18 +1339,23 @@ impl<'block> BrilligBlock<'block> {
             BrilligBinaryOp::LessThan,
         );
 
-        self.brillig_context.codegen_branch(result_is_negative.address, |ctx, is_negative| {
-            if is_negative {
-                // Two's complement of num
-                let zero = ctx.make_constant_instruction(0_usize.into(), num.bit_size);
-                ctx.binary_instruction(zero, num, absolute_value, BrilligBinaryOp::Sub);
-                ctx.deallocate_single_addr(zero);
-            } else {
-                // Simply move the original num
-                ctx.mov_instruction(absolute_value.address, num.address);
-            }
-        });
+        // Two's complement of num
+        let zero = self.brillig_context.make_constant_instruction(0_usize.into(), num.bit_size);
+        let twos_complement =
+            SingleAddrVariable::new(self.brillig_context.allocate_register(), num.bit_size);
```

Comment:
Isn't the old code better? Now we unconditionally execute all of absolute value (in the AVM prover it would be some extra rows). I know it reads nicer, but probably the old code is better unless we care about branch prediction (which I think we don't :P)

Reply:
Previously it was:
```
JUMPI then
JUMP else
CONST  // then
SUB
JUMP end
MOV // else
//end
```

Now it's
```
CONST
SUB
CMOV
```
So it's half the opcodes emitted
Regarding execution trace, previously, for the truthy case it was 4 opcodes, for the falsy case it was 3 opcodes. Now it's always 3 opcodes

I think now it's better! But it was mainly so CMOV was used and tested. The old case could probably have been optimized in a different way

Reply:
You are right!